HW04.Rmd

```diff
@@ -1,126 +1,343 @@
+gun_deaths <- read.csv("gun_deaths.csv")
+
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+
+```{r month, warning=FALSE}
+
+# generate new dataframe for this task
+gun_deaths_summary1 <- gun_deaths %>%
+    select(month) %>%
+    group_by(month) %>%
+    count(month) %>%
+    rename(deaths_per_month = n)
+
+# print results as a table using kable()
+kable(gun_deaths_summary1, format = "markdown", align = 'c', col.names = c("Month", "Deaths per Month"))
+
+
+```
+
+**If we further want to divide the number of gun deaths per month for each given year:**
+
+```{r month_2, warning=FALSE}
+
+# generate new dataframe, almost the same as above but also selecting and grouping by year
+gun_deaths_summary2 <- gun_deaths %>%
+    select(year, month) %>%
+    group_by(year, month) %>%
+    count(month) %>%
+    rename(deaths_per_month = n)
+
+# print results as a table using kable()
+kable(gun_deaths_summary2, format = "markdown", align = 'c', col.names = c("Year", "Month", "Deaths per Month"))
+
+
+```
+
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+
+# generate bar graph using the dataframe generated above
+ggplot(data = gun_deaths_summary1, aes(x = factor(month), y = deaths_per_month)) + # make sure month is plotted as factor so that we can convert to abbreviated month name
+    geom_col() +
+    scale_x_discrete(labels=month.abb) + # change x-axis labels to abbreviated month names instead of numbers
+    labs(title = "Number of Gun Deaths for each Month of the Year", subtitle = "Based on Data from 2012 - 2014", x = "Month", y = "Number of Deaths") + # adjust labels
+    theme_classic()   # add a theme (I like to use the pre-set theme theme_classic())
+
+
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+
+# generate new dataframe looking at the number of gun deaths associated with ech type of intent
+gun_deaths_intent <- gun_deaths %>%
+    select(year, month, intent) %>%
+    group_by(intent) %>%
+    count() %>%
+    rename(number_of_deaths = n)
+
+# print results as a table using kable()
+kable(gun_deaths_intent, format = "markdown", align = 'c', col.names = c("Intent", "Number of Deaths"))
+```
+
+**We can use this output to sort the x-axis accordingly:**
+
+```{r intent_2}
+
+# sort the x-axis based on table above
+gun_deaths_intent$intent <- factor(gun_deaths_intent$intent, levels = c("Suicide", "Homicide", "Accidental", "Undetermined", "Not Available"))
```

Comment:
I also found out that passing the x aesthetic as x = reorder(intent, -number_of_deaths) will sort the x-axis by descending values (without "-" it will be ascending). I imagine it being super useful when you have a lot of factors in your dataset.